dev/build-and-push-image.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+docker buildx build -t tabulario/spark-iceberg:latest -t tabulario/spark-iceberg:3.2.1_0.13.1 --platform=linux/amd64,linux/arm64 ./spark --push
```

Comment:
If we parameterize the tag we could maybe drive this using github tags or branch names, right? Also I think when you exclude the second `-t` it uses the tag name you gave it locally. What do you think of this?:
```
docker buildx build -t tabulario/spark-iceberg:$1 --platform=linux/amd64,linux/arm64 ./spark --push
```

Reply:
Yeah I'm good with that. I mostly threw this together just to have... something to include for our use.

I'd ideally like to check that things are installed and then most importantly check if the argument `$1`  exists. Possibly also that the image tag, outside of `latest` is unique (not sure if that's a goal).

I have some shell scripts that do more or less most of that already, I'll dig them up and then put them on this so we can go about automating the deployments on merge and/or on trigger (with tag specified then).

Reply:
I added in the tag argument, with validation.